rust/lance-encoding/src/data.rs

```diff
@@ -585,6 +700,21 @@ impl DataBlock {
             Self::Opaque(inner) => inner.num_values,
         }
     }
+
+    pub fn data_size(&self) -> u64 {
+        match self {
+            Self::AllNull(_) => 0,
+            Self::Nullable(inner) => inner.data_size(),
+            Self::FixedWidth(inner) => inner.data_size(),
+            Self::FixedSizeList(inner) => inner.data_size(),
+            Self::VariableWidth(inner) => inner.data_size(),
+            // not implemented yet
+            Self::Struct(_) => 0,
```

Comment:
Please use `todo!()` instead of `0`.  That way we don't accidentally forget we haven't done this.